src/volume_mgr.rs

```diff
@@ -680,36 +791,114 @@ where
 
     /// Consume self and return BlockDevice and TimeSource
     pub fn free(self) -> (D, T) {
-        (self.block_device, self.timesource)
+        (self.block_device, self.time_source)
+    }
+
+    /// Check if a file is at End Of File.
+    pub fn file_eof(&self, file: File) -> Result<bool, Error<D::Error>> {
```

Comment:
Honestly these feel like they should be members of `File` and accept the `Volume` as an arg?

Reply:
Hmm. I could take `&mut VolumeManager` as an arg in a File method, true. Its important you give the right VolumeManager but we can sort of check that now.

Reply:
This is #99